src/BenchmarkDotNet/Toolchains/CoreRun/CoreRunToolchain.cs

```diff
@@ -58,17 +61,27 @@ public class CoreRunToolchain : IToolchain
         public override string ToString() => Name;
 
         public bool IsSupported(BenchmarkCase benchmark, ILogger logger, IResolver resolver)
+            => !CheckSupport(benchmark, logger, resolver).Any();
+
+        public IEnumerable<ValidationError> CheckSupport(BenchmarkCase benchmark, ILogger logger, IResolver resolver)
         {
             if (!SourceCoreRun.Exists)
             {
-                logger.WriteLineError($"Provided CoreRun path does not exist, benchmark '{benchmark.DisplayInfo}' will not be executed. Please remember that BDN expects path to CoreRun.exe (corerun on Unix), not to Core_Root folder.");
-                return false;
+                var validationError = new ValidationError(true, $"Provided CoreRun path does not exist, benchmark '{benchmark.DisplayInfo}' will not be executed. Please remember that BDN expects path to CoreRun.exe (corerun on Unix), not to Core_Root folder.", benchmark);
+
+                logger.WriteLineError(validationError.Message);
+
+                return new List<ValidationError> { validationError };
```

Comment:
we can take advantage of yield return/break and simplify the code a bit:

```suggestion
                yield return new ValidationError(true, $"Provided CoreRun path does not exist, benchmark '{benchmark.DisplayInfo}' will not be executed. Please remember that BDN expects path to CoreRun.exe (corerun on Unix), not to Core_Root folder.", benchmark);
                yield break;
```

Reply:
Do you think it's worth it? Seems that it requires the same amount of code. I tend to avoid the yield keyword for finite collections, as it uses a state machine under the hood.

Reply:
I do. It simplifies the code + by returning `List` we return a mutable collection, which can be misused by some callers:

```cs
IEnumerable<ValidationError> collection = Validate(benchmark, resolver);
((List<ValidationError>)collection).Add(new ...)
```

In places where we know the size of the collection and it's fixed we can use arrays instead of `yield` or even `ImmutableArray` to make sure that the content never changes.